ProfilerGUI/Source/Configurator/MainViewModel.cs

```diff
@@ -0,0 +1,190 @@
+                    UiUtils.Raise(PropertyChanged, this, nameof(SelectedBitnessIndex));
+                }
+            };
+
+            uploadConfig = ReadUploadConfigFromDisk();
+        }
+
+        /// <summary>
+        /// Shows the dialog to configure the trace upload.
+        /// </summary>
+        internal void OpenUploadConfigDialog()
+        {
+            UploadConfigWindow dialog = new UploadConfigWindow(uploadConfig);
+            bool? result = dialog.ShowDialog();
+            if (result == true)
+            {
+                uploadConfig = dialog.Config;
+            }
+            UiUtils.Raise(PropertyChanged, this, nameof(UploadSummary));
+        }
+
+        /// <summary>
+        /// Resets the fields that describe the application that should be profiled.
+        /// </summary>
+        internal void ClearTargetAppFields()
+        {
+            TargetApp.Clear();
+        }
+
+        /// <summary>
+        /// Saves the current configuration to a file.
+        /// </summary>
+        internal void SaveConfiguration()
+        {
+            try
+            {
+                TargetApp.Configuration.WriteToFile();
+                logger.Info("Wrote config to {configPath}", ProfilerConfiguration.ConfigFilePath);
+                SaveUploadConfig();
+            }
+            catch (Exception e)
+            {
+                logger.Error(e, "Could not save configuration to {configPath}", ProfilerConfiguration.ConfigFilePath);
+            }
+        }
+
+        private UploadConfig ReadUploadConfigFromDisk()
+        {
+            if (!File.Exists(UploadConfigFilePath))
+            {
+                return null;
+            }
+
+            try
+            {
+                return UploadConfig.ReadFromFile(UploadConfigFilePath);
+            }
+            catch (Exception e)
+            {
+                logger.Error(e, "Failed to load upload deamon configuration from {uploadConfigPath}", UploadConfigFilePath);
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Saves the config for the upload daemon.
+        /// </summary>
+        private void SaveUploadConfig()
+        {
+            try
+            {
+                File.WriteAllText(UploadConfigFilePath, JsonConvert.SerializeObject(uploadConfig));
+                logger.Info("Wrote upload deamon config to {configPath}", UploadConfigFilePath);
+            }
+            catch (Exception e)
+            {
+                logger.Error(e, "Failed to save upload deamon configuration to {uploadConfigPath}", UploadConfigFilePath);
+            }
+        }
+
+        /// <summary>
+        /// Starts the profiled application with the profiler attached.
+        /// </summary>
+        internal void RunProfiledApplication()
+        {
+            if (string.IsNullOrEmpty(TargetApp.ApplicationPath))
+            {
+                logger.Error("You did not configure an application to profile");
```

Comment:
Be instructive? "Please configure/select an application..."

Naming inconsistency: "application" vs. "Select app" on the GUI button.